json_rpc/rpcproxy.nim

```diff
@@ -0,0 +1,38 @@
+import servers/[socketserver, httpserver]
+import clients/[socketclient, httpclient]
+
+type RpcHttpProxy* = ref object of RootRef
+    rpcHttpClient: RpcHttpClient
+    rpcHttpServer: RpcHttpServer
+
+proc proxyCall(client: RpcHttpClient): ProxyCall =
+  result = 
+    proc (name: string, params: JsonNode): Future[Response] = client.call(name, params)
+
+proc new*(T: type RpcHttpProxy, listenAddresses: openArray[string]): T = 
+    let client = newRpcHttpClient()
+    let router = RpcRouter.init()
+    T(rpcHttpClient: client, rpcHttpServer: newRpcHttpServer(listenAddresses, router))
+
+proc newRpcHttpProxy*(listenAddresses: openArray[string]): RpcHttpProxy =
+    RpcHttpProxy.new(listenAddresses)
+
+proc init*(proxy:RpcHttpProxy, proxyServerUrl: string) {.async.} =
```

Comment:
We use `init` for "object construction" of non ref objects, as we use `new` for ref objects, like you did correctly above. 

So I would avoid calling this `init` but instead prefer `start`